### ISSUE TITLE
sway 1.6.1 screen flickering after launching firefox

### ISSUE THREAD

**Issue by NobodyXu** [swaywm/sway#6408]
### Please fill out the following:
- **Sway Version: 1.6.1**

- **Debug Log: [gist link]**

- **Configuration File: [gist link]**

- **Description:**
    - I launched firefox with `MOZ_ENABLE_WAYLAN=1` inside the firejail sandbox and my screen frozen
    - I switched to console then switched back, then the screen become flickering but the screen was not frozen. I could see the background image in the flickering.
    - I tried disable the background image but didn’t help

Here is a [demo](https://youtu.be/YeWZEuY93vw) that is recorded separately:

[gist link]: https://gist.github.com/NobodyXu/8cc05272939235156b617a53d30ba0ae

**Comment by divoxx** [comment]
I'm also having similar issues after the upgrade to 1.6.1. Sway completely freezes and I get screen flickering and artifacts, but mine is the entire screen and not just a subsection like in the video. It feels very similar though.

It also happens after opening other apps, not just firefox, so it's likely not specific to it.

**Comment by vyivel** [comment]
According to the logs, you use wlroots 0.14.0. Try again with wlroots 0.14.1.

**Comment by NobodyXu** [comment]
@vyivel Sadly, the distro I am using [does not have wlroots 0.14.1 yet](https://packages.gentoo.org/packages/gui-libs/wlroots).

Is there other things I can try?

**Comment by NobodyXu** [comment]
> I'm also having similar issues after the upgrade to 1.6.1. Sway completely freezes and I get screen flickering and artifacts, but mine is the entire screen and not just a subsection like in the video. It feels very similar though.
> 
> It also happens after opening other apps, not just firefox, so it's likely not specific to it.

I found that I can actually run the terminal `foot` in the sway — It is completely OK.

**Comment by vyivel** [comment]
>Is there other things I can try?

You can try to compile sway yourself.
https://github.com/swaywm/sway/wiki/Development-Setup

**Comment by NobodyXu** [comment]
> > I'm also having similar issues after the upgrade to 1.6.1. Sway completely freezes and I get screen flickering and artifacts, but mine is the entire screen and not just a subsection like in the video. It feels very similar though.
> > It also happens after opening other apps, not just firefox, so it's likely not specific to it.
> 
> I found that I can actually run the terminal `foot` in the sway — It is completely OK.

I also found that opening google-chrome without firejail or telegram with firejail or `feh` through `bemenu-run`  or the terminal has the same effect.

**Comment by NobodyXu** [comment]
After creating 10 `foot` terminal in 10 separate workspaces and closes one of them, swaywm also freezes.

**Comment by NobodyXu** [comment]
It seems that copying and pasting for three times also freezes swaywm

**Comment by NobodyXu** [comment]
> > Is there other things I can try?
> 
> You can try to compile sway yourself.
> https://github.com/swaywm/sway/wiki/Development-Setup

Actually I was using gentoo, so I kindof compiled every packages on my machine.

**Comment by NobodyXu** [comment]
Here’s the `dmesg` output when I killed `swaywm`:

```
[ 2200.784316] nouveau 0000:01:00.0: DRM: Moving pinned object 00000000f8651179!                                                                                                      [97/2699][ 2200.790623] nouveau 0000:01:00.0: DRM: bo 00000000f8651179 pinned elsewhere: 0x00000001 vs 0x00000002
[ 2200.790925] nouveau 0000:01:00.0: DRM: bo 00000000f8651179 pinned elsewhere: 0x00000001 vs 0x00000002
... endless of pinned elsewhere
[ 2242.572668] ------------[ cut here ]------------                                                                                                                                    [6/2699][ 2242.572675] WARNING: CPU: 6 PID: 1 at 0xffffffffc054b0ae                                                                                                                                    [ 2242.572679] Modules linked in: kvm_intel kvm irqbypass snd_hda_codec_realtek iTCO_wdt iTCO_vendor_support watchdog iwlmvm iwlwifi snd_hda_codec_hdmi nouveau drm_ttm_helper ttm ucsi_acpi tpm_crb tpm_tis tpm_tis_core tpm squashfs coretemp                                                                                                                                               [ 2242.572691] CPU: 6 PID: 1 Comm: systemd Tainted: G        W         5.13.4-gentoo #4                                                                                                        [ 2242.572695] Hardware name: HP OMEN by HP Laptop 15-dc0xxx/84DB, BIOS F.09 06/11/2019                                                                                                        [ 2242.572698] RIP: 0010:0xffffffffc054b0ae                                                                                                                                                    [ 2242.572701] Code: 00 00 41 c6 44 24 08 00 c6 83 60 11 00 00 00 48 83 7d 08 00 74 08 48 89 ef e8 2e 4b e7 f5 5b 48 89 ef 5d 41 5c e9 82 b2 ac f5 <0f> 0b e9 6c ff ff ff 66 66 2e 0f 1f 84 00 00 00 00 00 41 56 41 55                                                                                                                                                                        [ 2242.572705] RSP: 0018:ffff9453c0043e48 EFLAGS: 00010206                                                                                                                                     [ 2242.572708] RAX: 000000000000005a RBX: ffff8fff03864800 RCX: 0000000000000000                                                                                                               [ 2242.572711] RDX: 0000000000000002 RSI: 0000000000000001 RDI: ffff8fff115a5800                                                                                                               [ 2242.572713] RBP: ffff8fff115a5800 R08: 0000000000000001 R09: ffff8fff00042700                                                                                                               [ 2242.572716] R10: ffff8fff5728ae00 R11: 0000000000000000 R12: ffff8fff057586d0
[ 2242.572719] R13: ffff8fff596e5000 R14: ffff8fff000aea20 R15: 0000000000000000
[ 2242.572722] FS:  00007fb05fb46f40(0000) GS:ffff90028db80000(0000) knlGS:0000000000000000
[ 2242.572725] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 2242.572728] CR2: 00007f3468000010 CR3: 00000002df5d0002 CR4: 00000000003706e0
[ 2242.572731] Call Trace:
[ 2242.572734]  ? 0xffffffffc054d95c
[ 2242.572736]  ? 0xffffffffb63ce661
[ 2242.572738]  ? 0xffffffffb642bdf5
[ 2242.572740]  ? 0xffffffffb60558de
[ 2242.572741]  ? 0xffffffffb603d3ff
[ 2242.572743]  ? 0xffffffffb5ed7bb7
[ 2242.572745]  ? 0xffffffffb5f21a50
[ 2242.572747]  ? 0xffffffffb6913cdd
[ 2242.572749]  ? 0xffffffffb6039358
[ 2242.572750]  ? 0xffffffffb69108fd
[ 2242.572752]  ? 0xffffffffb6a0007c
[ 2242.572754] ---[ end trace 3ea5492ca9555e32 ]---
```

**Comment by vyivel** [comment]
>Actually I was using gentoo, so I kindof compiled every packages on my machine.

I meant to compile it with wlroots HEAD and check if the issue was already fixed in master.
If not, try using `git bisect` to find the bad commit, assuming everything worked fine before.

**Comment by NobodyXu** [comment]
> > Actually I was using gentoo, so I kindof compiled every packages on my machine.
> 
> I meant to compile it with wlroots HEAD and check if the issue was already fixed in master.
> If not, try using `git bisect` to find the bad commit, assuming everything worked fine before.

I can have a try since my computer should have every dependency required for compiling sway.

**Comment by emersion** [comment]
The nouveau issues should be fixed with wlroots 0.14.1.

**Comment by NobodyXu** [comment]
> The nouveau issues should be fixed with wlroots 0.14.1.

Nice!

I will upgrade ASAP.

**Comment by NobodyXu** [comment]
Pulled in the trunk swaywm and wlroots as the development setup and compiled, the problem is solved.
It even fixed some old bug, like right click in firefox gives a flickering menu.

Thank you so much @vyivel @emersion !
